### PR TITLE
[PR] Dev branch Pull Request

### PR DESCRIPTION
**Pull request by hanulrheem9218**
Features:
- Added a score counter.
- Implemented a space button for player jump.
- Added a 2D physics to detect player collision.
- Added a random spawn generator for the obstacles.
- Added a horizontal scroller for the background.

### CODE REVIEW

File: Assets/2DSprite/FlappyBird (2).png
Comment (CheeseStick) [inline Assets/2DSprite/FlappyBird (2).png:1]: Is this file required? It seems like it's a duplicate of `FlappyBird.png`.

File: Assets/2DSprite/pipe2.png
Comment (CheeseStick) [inline Assets/2DSprite/pipe2.png:1]: Is this asset removed? I think rotating the `pipe.png` 180 deg can give the same result.

File: Assets/2DSprite/FlappyBird.png
Comment (CheeseStick) [inline Assets/2DSprite/FlappyBird.png:1]: nit: would be nice to unify the naming rule (or follow the standard). I can see most of the assets are using snake cases, but it's in camel case.

File: Assets/Scripts/IObstacles.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+public interface IObstacles
```
Comment (CheeseStick) [inline Assets/Scripts/IObstacles.cs]: nit: As it's an interface for a single obstacle, maybe rename it to `IObstacle`?

File: Assets/Scripts/IObstacles.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+{
+    public void MoveObstacle();
```
Comment (CheeseStick) [inline Assets/Scripts/IObstacles.cs]: non-blocking: Because it's under `IObstacle`, maybe this doesn't need to repeat what it moves. Is the `Move` in use from somewhere? if not, Can it be updated to `Move`?

File: Assets/Scripts/Obstacle.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+using System.Collections;
+using System.Collections.Generic;
+using Unity.VisualScripting;
+using UnityEngine;
+
+public class Obstacle : MonoBehaviour //, IObstacles
```
Comment (CheeseStick) [inline Assets/Scripts/Obstacle.cs:6]: nit: If interface `IObstacles` is not in use, I suggest removing it to reduce the confusion.

File: Assets/Scripts/ObstacleHolder.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    public void InitiateRandomPosition()
+    {
+        Vector3 newPosition = new Vector3(transform.position.x, -(Random.Range(3.3f, 4.5f)), transform.position.z);
+        transform.position = newPosition;
+    }
+
+    public void MoveObstacle()
+     {
+         transform.position += Time.deltaTime * (Vector3.left *speed);
```
Comment (CheeseStick) [inline Assets/Scripts/ObstacleHolder.cs]:
```suggestion
         transform.position += Time.deltaTime * (Vector3.left * speed);
```

File: Assets/Scripts/ObstacleHolder.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ObstacleHolder : MonoBehaviour, IObstacles
+{
+    [SerializeField,Header("Obstacle movement speed"),Range(0,10)]
+    public float speed = 3f;
```
Comment (CheeseStick) [inline Assets/Scripts/ObstacleHolder.cs]:
```suggestion
    public float speed = 3.0f;
```

File: Assets/Scripts/ObstacleHolder.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        transform.position = newPosition;
+    }
+
+    public void MoveObstacle()
+     {
+         transform.position += Time.deltaTime * (Vector3.left *speed);
+     }
+    void Start()
+    {
+        if (speed == 0) speed = 3f;
```
Comment (CheeseStick) [inline Assets/Scripts/ObstacleHolder.cs]:
```suggestion
        if (speed == 0) { speed = 3.0f };
```

File: Assets/Scripts/UIManager.cs [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    void Start()
+    {
+        _scoreText = gameObject.transform.Find("ScoreDisplayer").GetComponent<TextMeshProUGUI>();
+        _displayDeath = gameObject.transform.Find("DeathScreen").gameObject;
+        _displayDeath.SetActive(false); 
+    }
+
+    public void IncreaseScore()
+    {
+        _scoreText.text = $"{++_scoreCount}";
```
Comment (CheeseStick) [inline Assets/Scripts/UIManager.cs]:
```suggestion
        _scoreCount += 1;
        _scoreText.text = $"{_scoreCount}";
```

**Review by CheeseStick — APPROVED** [review]

File: Assets/2DSprite/FlappyBird (2).png
Comment (hanulrheem9218) [reply Assets/2DSprite/FlappyBird (2).png:1]: I have checked the file, and you were correct; the file was necessary.

File: Assets/2DSprite/pipe2.png
Comment (hanulrheem9218) [reply Assets/2DSprite/pipe2.png:1]: Yes, it has been removed now.

File: Assets/2DSprite/FlappyBird.png
Comment (hanulrheem9218) [reply Assets/2DSprite/FlappyBird.png:1]: I have changed their names to be camel case, thanks for the reminder.

File: Assets/Scripts/IObstacles.cs (hunk 1 above)
Comment (hanulrheem9218) [reply Assets/Scripts/IObstacles.cs]: like you mentioned to me about the naming convention,  i have now named it as "ObstacleController".

Comment (hanulrheem9218) [reply Assets/Scripts/IObstacles.cs]: and removed IObstacle interface class.
